<a href="https://colab.research.google.com/github/Yahia0mohamed/TECHNOHACKS_ML_intern/blob/main/Diabetes_Prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# name: Yahia mohamed
**TECHNOHACKS ML Intern//**   project: Diabetes Prediction 'TASK 3'

this model aims to predict if a person has diabetes or
not.

In [ ]:
#importing my libraries
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score

importing the DataSet and get some of it's features

In [ ]:
Dia_DF=pd.read_csv('diabetes.csv')

In [ ]:
Dia_DF.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

0 ----> 'don't have diabetes'

1 ----> 'have diabetes'

In [ ]:
Dia_DF.shape
Dia_DF['Outcome'].value_counts()

0    500
1    268
Name: Outcome, dtype: int64

spliting lable from features


In [ ]:
y=Dia_DF['Outcome']
x=Dia_DF.drop(columns='Outcome',axis=1)

**Train , Test split**

In [ ]:
# train,test split technique
X_train,X_test,Y_train,Y_test=train_test_split(x,y,test_size=0.2,stratify=y,random_state=1)

Train the model ----> Logistic Regression Model

In [ ]:
myModel=LogisticRegression(max_iter=1000)

In [ ]:
myModel.fit(X_train,Y_train)

LogisticRegression(max_iter=1000)

In [ ]:
pred=myModel.predict(X_test)
accu= accuracy_score(pred,Y_test)
print(accu*100)

78.57142857142857


so our Model accuercy is = 78.57%

system to use our model in

In [ ]:
myInputTest=(3,126,88,41,235,39.3,0.704,27)
z=np.asarray(myInputTest)

modelInput= z.reshape(1,-1)

prediction= myModel.predict(modelInput)

if(prediction[0]==0):
  print('has no diabetes')
else:
  print('has diabetes')


no diabetes


/usr/local/lib/python3.10/dist-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but LogisticRegression was fitted with feature names
  warnings.warn(
